src/microdot.py

```diff
@@ -228,11 +228,11 @@ def __init__(self, app, client_addr, method, url, http_version, headers,
         self.content_length = 0
         self.content_type = None
         for header, value in self.headers.items():
-            if header == 'Content-Length':
+            if header.lower() == 'content-length':
                 self.content_length = int(value)
-            elif header == 'Content-Type':
+            elif header.lower() == 'content-type':
                 self.content_type = value
-            elif header == 'Cookie':
+            elif header.lower() == 'cookie':
```

Comment:
Here you are converting each header to lower case three times. This should be done once.